cbor/src/main/java/com/fasterxml/jackson/dataformat/cbor/CBORGenerator.java

```diff
@@ -402,9 +417,20 @@ public final void writeStringField(String fieldName, String value)
             return;
         }
         _verifyValueWrite("write String value");
+        _decrementList();
         _writeString(value);
     }
 
+	
+    public final void writeFieldLong(long size) throws IOException {
```

Comment:
So, would the intent here be to allow writing long-keyed Maps/Objects? Since this would not be exposed via Jackson streaming API, the only way to use this would be with custom serializers. But at very least Javadocs would have to explain the method (including where it was added); this would have to be in 2.8 as it's new part of API.
I am not 100% sure if it makes sense to add format-specific extensions, but I guess that could be acceptable, as some format modules have something similar (I think CSV at least, perhaps XML).